### DISCUSSION TITLE
Pelican-themes is falling apart. How can we help?

### DISCUSSION THREAD

**Discussion by maphew** [getpelican/pelican#3089] [Q&A]
The [Pelican-themes][PT] repository and associated  material are falling apart. Preview site broken (probably irreperable; 3rd party dependence). Duplicate unclosed issues. Open issues which are actually finished and should be closed, or are so stale they don't matter any more. Double digit numbers of completely broken themes, many others that need tweaks. Long standing open pull requests of theme fixes and new ones. It all paints a depressing picture of the state of the Pelican project as a whole, one that is on it's way to irrelevance and abandonment in a not far off future. However looking at the commit rate in the main Pelican repository, and the number of sites out there in the wild that using Pelican (new ones too) I don't think this glum view is necessary. So lets wipe off the grime, sand a few things, patch a couple holes, and throw on a fresh coat of paint! 

I've put in some effort [(o)][0] to revitalize but perhaps not on the parts that actually need it. So hey @justinmayer, @almet: I'm loaded and ready to go! Point me where I can do some good. ;-) I bet others will pitch in too if there is a little direction, and together we can build a sense of progress to goals.

[PT]: https://github.com/getpelican/pelican-themes
[0]: https://github.com/maphew/pelican-themes-preview/blob/main/README.md

**Comment by galaxy4public** [comment]
I second this:
>  It all paints a depressing picture of the state of the Pelican project as a whole

It is not only themes, but the plugins infrastructure too.  I am willing to volunteer some time to help out with bringing it back to a manageable state, if needed.  Overall, it is really painful to observe how the environment of, in my opinion, one of the best static website generators is rapidly deteriorating (most likely due to lack of spare time of the current maintainers).

**Reply by maphew** [reply]
Thanks for commenting @galaxy4public. I had considered deleting the thread since an empty open question would add gloom to the aura! Since starting it I've also had occasion to run into the similar plugins situation, though that looks a bit more lively. That makes sense to me, attend to function before looks. I've also made a suggestion [there](https://github.com/getpelican/pelican-plugins/issues/1379).

> likely due to lack of spare time of the current maintainers

Probably exactly it, and I imagine a thread like this adds to sense of dread and time scarcity because "what can we do" is an open ended question, asking for yet more time to think and respond to. I guess for now we keep walking through the thickets and wedging aside the branches as best we can until something specific makes itself known.

**Reply by galaxy4public** [reply]
@maphew , I would argue that at least @justinmayer is a very active project maintainer and an is quite responsive, so I personally don't mind having such a thread open as a destination for quick suggestions from the maintainers on where the help is needed.  I do not consider it to be gloomy since it shows that the community is alive and cares.  Indeed, this is just my opinion, though -- you are the topic starter, so if you think it is better to close the discussion it is entirely up to you.  On the other hand, I think if people who have time to volunteer subscribe to this discussion, it can be like an announcement platform to share when and where the help is needed.

**Reply by maphew** [reply]
There's two of us, it's alive. ;-)
In the meantime I found a couple things in Plugins I could "just do", so I did: https://github.com/getpelican/pelican-plugins/pull/1381

**Comment by maphew** [comment]
Would the idea behind the Pelican-plugins organization instead of Getpelican/pelican-plugins repo also work for themes? I'm not too familiar with the Pelican universe yet but from what I've seen the answer is yes. I put my name forward to hep get such an effort started. I can't promise to stay fully engaged for years but I can certainly put in a fair bit of work over the next few months to help get it standing.

**Comment by datatalking** [comment]
@maphew @justinmayer @galaxy4public I'm facing this as well, while its not the same as pelican, the design, responsiveness and intuitive design behind [Hugo](https://themes.gohugo.io/themes/docuapi/) could give us some design choices.

I would love to build my static site in Pelican, and got the tutorial running but Pelican needs help and I'm offering to contribute. LMK

**Comment by minchinweb** [comment]
It would be a fairly substantial change, but would it make sense to package (all) Pelican theme as pip-install-able packages? That's what I did for my own theme [seafoam](https://github.com/MinchinWeb/seafoam) ([details on how it works](https://blog.minchin.ca/2021/07/seafoam-260.html)). There is a PyPI Identifier for [Pelican Themes](https://pypi.org/search/?q=&o=&c=Framework+%3A%3A+Pelican+%3A%3A+Themes), and so that could be used as a way to find them.

(None of that helps the `pelican-theme` repo, but maybe it's still the way forward.)

That said, I very much like the idea of having a standardize "test site" to run themes against, and a gallery of the results. I look forward to helping, if there way an obvious way to do so.

**Reply by Epy** [reply]
Like https://www.csszengarden.com/ :)

**Reply by avaris** [reply]
> It would be a fairly substantial change, but would it make sense to package (all) Pelican theme as pip-install-able packages?

Short answer, yes. Longer answer, yes but requires quite a bit of work and by extension someone to do said work :). Your current method of masking a theme as plugin, although clever, not a viable solution. Makes it very cumbersome to test themes, if you rely on autodiscovery of plugins. You'd need to install-uninstall-install-uninstall-... It would need to be in its own namespace.

On the other hand, pip-installable themes makes "clone-tinker-customize" methodology of making your own theme based on another slightly less beginner friendly, at least for people not familiar with python packaging.

**Comment by johann-petrak** [comment]
Is there currently any other way to browse existing Pelican themes visually in order to select the one that fits my aims best? 
I am not a HTML/CSS designer so I have to rely on finding a good theme that works on both desktop and mobile devices, but currently it looks very hard to do this: there is no way to search for themes with specific properties, apparently no way to browse themes visually or anything that would make it easy to find a good theme. 
This is really the only but crucial part that makes Jekyll look like the more attractive way to go forward.

**Reply by justinmayer** [reply]
_Update:_ https://pelicanthemes.com is now back up. Many thanks to @maphew and @avaris for their work on this! 🎊 ~~Until a longer-term solution is in place, try this: https://avaris.github.io/pelican-themes/~~

**Reply by johann-petrak** [reply]
Brilliant, thank you, this is exactly what I was looking for!
